swim/memberlist.go

```diff
@@ -290,9 +291,16 @@ func (m *memberlist) SetLocalLabel(key, value string) error {
 		m.local.Labels = make(map[string]string)
 	}
 
+	old, had := m.local.Labels[key]
+
 	// set the label
 	m.local.Labels[key] = value
-	m.postLocalUpdate()
+
+	if !had || old != value {
```

Comment:
Do you think we should return a `bool` here to indicate a change (like on removal)?